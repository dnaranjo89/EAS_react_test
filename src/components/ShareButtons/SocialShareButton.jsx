import React from 'react';
import PropTypes from 'prop-types';
import {
  FacebookShareButton,
  TwitterShareButton,
  TelegramShareButton,
  WhatsappShareButton,
  EmailShareButton,
  FacebookIcon,
  TwitterIcon,
  TelegramIcon,
  WhatsappIcon,
  EmailIcon,
} from 'react-share';

import classnames from 'classnames/bind';
import STYLES from './SocialShareButton.scss';

const c = classnames.bind(STYLES);

const SocialButton = ({ url, type, size, className }) => {
  const buttonProps = { url, className };
  switch (type) {
    case 'whatsapp':
      return (
        <WhatsappShareButton {...buttonProps}>
          <WhatsappIcon size={size} />
        </WhatsappShareButton>
      );
    case 'facebook':
      return (
        <FacebookShareButton {...buttonProps}>
          <FacebookIcon size={size} />
        </FacebookShareButton>
      );
    case 'twitter':
      return (
        <TwitterShareButton {...buttonProps}>
          <TwitterIcon size={size} />
        </TwitterShareButton>
      );

    case 'telegram':
      return (
        <TelegramShareButton {...buttonProps}>
          <TelegramIcon size={size} />
        </TelegramShareButton>
      );
    case 'email':
      return (
        <EmailShareButton {...buttonProps}>
          <EmailIcon size={size} />
        </EmailShareButton>
      );
    default:
      return null;
  }
};

SocialButton.propTypes = {
  url: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  size: PropTypes.number,
  className: PropTypes.string.isRequired,
};

SocialButton.defaultProps = {
  size: 32,
};

const SocialShareButton = ({ url, type }) => (
  <SocialButton url={url} type={type} className={c('SocialShareButton')} />
);

SocialShareButton.propTypes = {
  url: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
};

export default SocialShareButton;
