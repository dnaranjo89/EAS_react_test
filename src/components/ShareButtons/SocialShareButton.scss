@import '../variables';

.SocialShareButton {
  display: inline-block;
  margin: $spacing-xs;
  border-radius: .25rem;
  cursor: pointer;
  overflow: hidden;
}
