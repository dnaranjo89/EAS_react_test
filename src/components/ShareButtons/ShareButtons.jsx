import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import classnames from 'classnames/bind';
import SocialShareButton from './SocialShareButton.jsx';
import STYLES from './ShareButtons.scss';

const c = classnames.bind(STYLES);

const availableShareButtons = ['facebook', 'twitter', 'telegram', 'whatsapp', 'email'];

const ShareButtons = ({ sectionTitle, drawType, url, types }) => (
  <div className={c('ShareButtons')}>
    {sectionTitle && (
      <Typography variant="subtitle2" align="center">
        {sectionTitle}
      </Typography>
    )}
    {types
      .filter(type => availableShareButtons.indexOf(type) >= 0)
      .map(type => (
        <SocialShareButton key={type} url={url} drawType={drawType} socialType={type} />
      ))}
  </div>
);

ShareButtons.propTypes = {
  sectionTitle: PropTypes.string,
  drawType: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  types: PropTypes.arrayOf(PropTypes.string),
};
ShareButtons.defaultProps = {
  sectionTitle: '',
  types: ['facebook', 'twitter', 'telegram', 'whatsapp'],
};

export default ShareButtons;
