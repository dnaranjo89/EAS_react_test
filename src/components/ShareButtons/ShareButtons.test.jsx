import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShareButtons from './ShareButtons.jsx';

describe('ShareButtons', () => {
  it('should render all buttons by default', () => {
    const wrapper = shallow(<ShareButtons url="the_cool_url" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should only render the specified types', () => {
    const wrapper = shallow(<ShareButtons url="the_cool_url" types={['facebook']} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders the title if given', () => {
    const wrapper = shallow(<ShareButtons sectionTitle="The title" url="the_cool_url" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
