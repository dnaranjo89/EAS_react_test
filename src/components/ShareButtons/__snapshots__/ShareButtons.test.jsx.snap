// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShareButtons renders the title if given 1`] = `
<div>
  <WithStyles(Typography)
    align="center"
    variant="subtitle2"
  >
    The title
  </WithStyles(Typography)>
  <SocialShareButton
    key="facebook"
    type="facebook"
    url="the_cool_url"
  />
  <SocialShareButton
    key="twitter"
    type="twitter"
    url="the_cool_url"
  />
  <SocialShareButton
    key="telegram"
    type="telegram"
    url="the_cool_url"
  />
  <SocialShareButton
    key="whatsapp"
    type="whatsapp"
    url="the_cool_url"
  />
  <SocialShareButton
    key="email"
    type="email"
    url="the_cool_url"
  />
</div>
`;

exports[`ShareButtons should only render the specified types 1`] = `
<div>
  <SocialShareButton
    key="facebook"
    type="facebook"
    url="the_cool_url"
  />
</div>
`;

exports[`ShareButtons should render all buttons by default 1`] = `
<div>
  <SocialShareButton
    key="facebook"
    type="facebook"
    url="the_cool_url"
  />
  <SocialShareButton
    key="twitter"
    type="twitter"
    url="the_cool_url"
  />
  <SocialShareButton
    key="telegram"
    type="telegram"
    url="the_cool_url"
  />
  <SocialShareButton
    key="whatsapp"
    type="whatsapp"
    url="the_cool_url"
  />
  <SocialShareButton
    key="email"
    type="email"
    url="the_cool_url"
  />
</div>
`;
